botocore/stub.py

```diff
@@ -362,3 +388,13 @@ def _validate_response(self, operation_name, service_response):
             raise ParamValidationError(
                 report=(
                     "Service response should only contain ResponseMetadata."))
+
+
+class StubRequest(object):
+    """
+    A tracked request that was made to a stubbed client. Has two attributes,
+    ``operation_name`` and ``params``.
+    """
+    def __init__(self, operation_name, params):
+        self.operation_name = operation_name
```

Comment:
These shouldn't be settable after creation, I would rather have them as read-only properties. I would also like for the service name to be added in so that if we create a stubber that works on the session we won't have breakages when adding service name.